FormidableLabs/spectacle
# v9: Typescript Conversion

### Description

Converts codebase to TS.
Attempts to minimize functional changes

Only converted the `/src` folder; did not convert the `/docs` folder.

#### Type of Change

Please delete options that are not relevant (including this descriptive text).

- [x] Chore (refactor, optimization)

### How Has This Been Tested?

Test the app for any regressions!  No changes should be visible.

Steps to test:
1. Checkout branch, `yarn install` of course
2. Run `yarn checks` and verify all pass
3. Run `yarn start:js` and verify that the sample deck works without any differences
4. Run `yarn start:md` and verify nothing has changed

Additional testing with `spectacle-visual-editor`:
1. Checkout `spectacle-visual-editor` as a sibling to `spectacle`
2. In `spectacle`, run `yarn link`  and `yarn build`
3. Clear out `node_modules` from spectacle (this might be required to avoid conflicts with `styled-components`)
4. Switch to `spectacle-visual-editor` and run `yarn link spectacle` then `yarn install`
5. Run `yarn start` to bring up the visual editor
6. Ensure the editor works as expected
7. Run `yarn typecheck` and verify types are working
